pkg/routeagent/controllers/route/driver.go

```diff
@@ -59,3 +60,23 @@ func toggleCNISpecificConfiguration(iface string) error {
 	}
 	return nil
 }
+
+func (r *Controller) annotateNodeWithCNIInterfaceIP(nodeName string) error {
+	node, err := r.clientSet.CoreV1().Nodes().Get(nodeName, metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("unable to get node info for node %v, err: %s", nodeName, err)
+	} else {
+		annotations := node.GetAnnotations()
+		if annotations == nil {
+			annotations = map[string]string{}
+		}
+		annotations[CniInterfaceIp] = r.cniIface.ipAddress
+		node.SetAnnotations(annotations)
+		_, err := r.clientSet.CoreV1().Nodes().Update(node)
+		if err != nil {
+			return fmt.Errorf("error updatating node %q, err: %s", nodeName, err)
+		}
+		klog.V(log.DEBUG).Infof("Successfully annotated node %q with cniIfaceIP %q", nodeName, r.cniIface.ipAddress)
+	}
+	return nil
```

Comment:
To be safe, I think this should be wrapped in `retry.RetryOnConflict`.

Reply:
Done. Thanks for reviewing @tpantelis